src/OpenTK.Windowing.Common/Input/JoystickState.cs

```diff
@@ -15,7 +15,7 @@ namespace OpenTK.Windowing.Common.Input
     /// <summary>
     /// Encapsulates the state of a joystick device.
     /// </summary>
-    public struct JoystickState : IEquatable<JoystickState>
+    public struct JoystickState : IEquatable<JoystickState>, ICloneable
```

Comment:
Not the biggest fan of the `IClonable` interface due to the boxing allocations it will make when dealing with `struct`s.
See other review comment for alternative.

Reply:
I wrote a long essay explaining why ICloneable sucks, but apparently it got lost. Bottom line is that its usage is strongly discouraged, which is why we don't have ICloneable<T>. Let's definitely remove this interface.